src/intel_perfmon_uncore_discovery.c

```diff
@@ -559,12 +563,13 @@ int perfmon_uncore_discovery(int model, PerfmonDiscovery** perfmon)
                         switch (model)
                         {
                             case SAPPHIRERAPIDS:
+                            case EMERALDRAPIDS:
                             case GRANITERAPIDS:
                                 if (unit.box_type == SPR_DEVICE_ID_CHA)
                                 {
                                     cur->units[idx].filter_offset = 0xE;
                                 }
-                                else if (unit.box_type == SPR_DEVICE_ID_iMC || (unit.box_type == SPR_DEVICE_ID_HBM && model == SAPPHIRERAPIDS))
+                                else if (unit.box_type == SPR_DEVICE_ID_iMC || (unit.box_type == SPR_DEVICE_ID_HBM && model == SAPPHIRERAPIDS || model == EMERALDRAPIDS||model == GRANITERAPIDS))
```

Comment:
This line raises a compiler warning in regards to `||` and `&&` precedence. Is this line supposed to mean:

```c
((unit.box_type == SPR_DEVICE_ID_HBM && model == SAPPHIRERAPIDS) || model == EMERALDRAPIDS||model == GRANITERAPIDS)
```
or
```c
(unit.box_type == SPR_DEVICE_ID_HBM && (model == SAPPHIRERAPIDS || model == EMERALDRAPIDS||model == GRANITERAPIDS))
```
?

Reply:
That slipped through my review. The whole part ` || model == EMERALDRAPIDS||model == GRANITERAPIDS` is not required. Only the `SAPPHIRERAPIDS` might have a HBM unit.